## SPONGE CONFIG
/**:
  ros__parameters:
    motor_state_topic: "/state/motors"
    motor_thresholds: [100, 3995]
    measured_norm_topic: "/state/norm_forces"
    sensor_state_topic: "/state/sensors"
    desired_norm_forces_topic: "/cmd/norm_forces"
    desired_position_topic: "/cmd/positions"
    measured_velocity_topic: "/cmd/velocities"
    open_service_name: "open"
    close_service_name: "close"
    proportional_service_name: "proportional_activation"
    integrator_service_name: "integrator_activation"
    motor_sensor_mappings: ['35:0', '36:1', '37:2', '38:3,4']
    calibrate_service_name: "calibrate_sensors" 


proportional_controller:
  ros__parameters:
    motor_ids: [35, 36, 37, 38]
    sensor_weight_mappings: ["0:1", "1:1", "2:1", "3:1", "4:1"]
    gain: 600.0
    gain_service_name: "set_gain"


euler_integrator_controller:
  ros__parameters:
    dt: 0.001
    motor_ids: [35, 36, 37, 38]


close:
  ros__parameters:
    threshold: 0.1
    initial_velocity: 300.0
    motor_ids: [35, 36, 37, 38]

open:
  ros__parameters:
    motor_position: [100.0, 100.0, 100.0, 100.0]
    motor_ids: [35, 36, 37, 38]

    
slipping_avoidance:
  ros__parameters:
    coefficients: [0.0, 3.3, 0.0, 0.0, 0.0] # 0.3 friction coefficient  NON SERVE QUI
    slipping_service_name: "slipping_activation"
    delta_forces_topic: "delta_forces" 


hand_driver:
  ros__parameters:
    motor_ids: [31, 32, 33, 34, 35, 36, 37, 38]
    millisecondsTimer: 2
    baudrate: 1000000
    serial_port: "/dev/ttyUSB0"
    protocol_version: 2.0

fingertip_sensors:
  ros__parameters:
    millisecondsTimer: 2
    baudrate: 1000000
    serial_port: "/dev/ttyUSB1"
